### PR TITLE
Throw on duplicate metrics

### PR DESCRIPTION
**Pull request by SimenB**
Prometheus isn't able to parse it, no reason not to fail hard

### CODE REVIEW

**Comment by siimon** [comment]
Ah, didn't know Prometheus threw an error on duplicate names - thanks.

File: test/summaryTest.js [hunk 1]
```diff
@@ -63,6 +68,7 @@ describe('summary', function() {
 	});
 
 	it('should correctly use calculate other percentiles when configured', function() {
+		register.clear();
```
Comment (siimon) [inline test/summaryTest.js]: Do you need to clear here if it's done in the afterEach? 

Same on line 95

**Comment by SimenB** [comment]
Says something along the lines of `Unknown format`, IIRC

**Comment by SimenB** [comment]
https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

Can't find anything about it erroring, but it did for us...

File: test/summaryTest.js (hunk 1 above)
Comment (SimenB) [inline test/summaryTest.js]: Yeah, it creates a new one with the same name as in `beforeEach`

**Comment by siimon** [comment]
Yeah I tried it locally and I got an error from the scrape so you are right..

File: test/summaryTest.js (hunk 1 above)
Comment (siimon) [inline test/summaryTest.js]: wouldn't it be easier to change the name in the created Summary?

Comment (siimon) [inline test/summaryTest.js]: never mind, I changed my mind - it's better to clear else there is a risk we will by mistake have the same name in another test.

**Pull request merged**